File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/data/repository/BluetoothRepositoryImpl.kt
package com.maktubcompany.easy_blue_printer.plugin.data.repository

import com.maktubcompany.easy_blue_printer.plugin.data.datasource.BluetoothDataSource
import com.maktubcompany.easy_blue_printer.plugin.domain.entities.BluetoothDeviceEntity
import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository

class BluetoothRepositoryImpl(
    private val dataSource: BluetoothDataSource
) : BluetoothRepository {

    override fun getPairedDevices(): List<BluetoothDeviceEntity> {
        return dataSource.getPairedDevices()
    }

    override fun connectToDevice(address: String): Boolean {
        return dataSource.connectToDevice(address)
    }

    override fun disconnectToDevice(): Boolean {
        return dataSource.disconnectToDevice()
    }

    override fun printData(data: String, size: Int, align: Int, bold: Boolean): Boolean {
        return dataSource.printData(data, size, align, bold)
    }

    override fun printEmptyLine(callTimes: Int): Boolean {
       return dataSource.printEmptyLine(callTimes)
    }

    override fun isConnected(): Boolean {
        return dataSource.isConnected()
    }

    override fun printImage(data: ByteArray, align: Int): Boolean {
        return dataSource.printImage(data, align)
    }
}

File: ios/Classes/plugin/data/repositories/BluetoothRepositoryImpl.swift
import Foundation

// Implementação do repositório Bluetooth
public class BluetoothRepositoryImpl: BluetoothRepository {
    private let dataSource: BluetoothDataSource

    // Inicializador
    public init(dataSource: BluetoothDataSource) {
        self.dataSource = dataSource
    }

    // Escanear e retornar dispositivos Bluetooth disponíveis
    public func scanDevices() -> [BluetoothDeviceEntity] {
        return dataSource.scanDevices()
    }

    // Conectar ao dispositivo pelo endereço
    public func connectToDevice(address: String) -> Bool {
        return dataSource.connectToDevice(address: address)
    }

    // Desconectar do dispositivo atualmente conectado
    public func disconnectFromDevice() -> Bool {
        return dataSource.disconnectFromDevice()
    }

    // Imprimir dados com tamanho, alinhamento e opção de negrito
    public func printData(data: String, size: Int, align: Int, bold: Bool) -> Bool {
        return dataSource.printData(data: data, size: size, align: align, bold: bold)
    }

    // Imprimir linha em branco um número específico de vezes
    public func printEmptyLine(callTimes: Int) -> Bool {
        return dataSource.printEmptyLine(callTimes: callTimes)
    }
}
